src/ColumnSelectionStep/ColumnSelectionStep.tsx

```diff
@@ -97,21 +97,25 @@ function GeneralizationControls({
           <Form.Item label="Substring start" name="substringStart">
             <span key={column.resetCount}>
               <InputNumber
+                // `precision` and `Math.round` in change handler both needed to protect from decimals
+                precision={0}
                 size="small"
                 placeholder="1"
                 min={1}
                 value={column.substringStart as number}
```

Comment:
Why does `value` work here but not in threshold selection? Weird...

Reply:
I would be guessing that the difference comes from the different context of rendering these components, but I have no idea how so. It seems `Form` is quite messy, I've found lots of threads and issues related to thing not working as expected there.